db/point_accrual.py

```diff
@@ -28,6 +29,69 @@ def get_multiplier_for_user(roles: list[Role]) -> int:
             return multiplier
     return 1
 
+def accrue_morning_points(
+    user_id: int, session: sessionmaker
+) -> bool:
+    """Accrues morning greeting points for a given user
+
+    Args:
+        user_id (int): Discord user ID to give points to
+        session (sessionmaker): Open DB session
+
+    Returns:
+        bool: True if points were awarded to the user
+    """
+    with session() as sess:
+        result = sess.execute(
+            select(MorningPoints).where(MorningPoints.user_id == user_id)
+        ).first()
+        if result is None:
+            sess.execute(
+                insert(MorningPoints).values(user_id=user_id, weekly_count=1)
+            )
+            return True
+
+        # Ensure points are not accruing on every message
+        # Only award points once per stream
+        morning_points: MorningPoints = result[0]
+        last_accrued: datetime = morning_points.timestamp
+        now = datetime.now()
+        time_difference = now - last_accrued
+
+        if time_difference < MORNING_DELTA:
+            return False
+        updated_timestamp = now
+
+        sess.execute(
+            update(MorningPoints)
+            .where(MorningPoints.user_id == user_id)
+            .values(
+                points=morning_points.weekly_count + 1,
+                timestamp=updated_timestamp,
+            )
+        )
+        return True
+
+def get_morning_points(user_id: int, session: sessionmaker) -> int:
+    """Get the number of morning greetings a user has accrued
+
+    Args:
+        user_id (int): Discord user ID to give a morning greeting to
+        session (sessionmaker): Open DB session
+
+    Returns:
+        int: Number of morning greetings currently awarded
+    """
+    with session() as sess:
+        result = sess.execute(
+            select(MorningPoints).where(MorningPoints.user_id == user_id)
+        ).first()
+
+        if result is None:
+            return 0
+
+        morning_points: MorningPoints = result[0]
+        return morning_points.weekly_count
```

Comment:
Since this query is going to be run fairly often, can we outdent this so the session closes ASAP?